packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -113,6 +115,34 @@ function validateLightDomTemplate(template: Template, vm: VM) {
     }
 }
 
+export function parseFragment(strings: string[], ...keys: number[]): () => Element {
+    return function (): Element {
+        const {
+            context: { hasScopedStyles, stylesheetToken },
+            shadowMode,
+        } = getVMBeingRendered()!;
+        const classToken = hasScopedStyles && stylesheetToken ? stylesheetToken : '';
+        const attrToken =
+            stylesheetToken && shadowMode === ShadowMode.Synthetic ? stylesheetToken : '""';
+
+        const htmlFragments: string[] = [];
+        for (let i = 0, n = keys.length; i < n; i++) {
+            switch (keys[i]) {
+                case 0:
+                    htmlFragments.push(strings[i], classToken);
+                    break;
+                case 1:
+                    htmlFragments.push(strings[i], attrToken);
+                    break;
+            }
+        }
+
+        htmlFragments.push(strings[strings.length - 1]);
```

Comment:
What is this extra `push` for?

Reply:
because template strings are processed in a way that it produces two chunks, he is joining the two chunks and a remaining index is pending to be added at the end.